src/components/Molecules/CtaPrompt/CtaPrompt.component.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * @file CtaPrompt.component.js
+ * Exports a CTA Prompt component.
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import CtaMessage from '../CtaMessage/CtaMessage.component';
+
+const cookiePrefix = 'cta_prompt_';
+
+/**
+ * Component that renders a CTA (Call to action) prompt.
+ */
+export default class CtaPrompt extends Component {
+  static propTypes = {
+    children: PropTypes.node,
+    type: CtaMessage.propTypes.type
+  };
+
+  static defaultProps = {
+    children: [],
+    type: null
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      shownMessage: null
+    };
+
+    if (props.children) {
+      const children = (props.children.length
+        ? props.children
+        : [props.children]
+      ).filter(child => child.type === CtaMessage);
+
+      // Determine which (if any) message should be shown.
+      children.forEach(msg => {
+        const { data } = msg.props;
+
+        const hash = this.getCookie(data.name);
+
+        // Show first message that either:
+        // - Doesn't have a cookie.
+        // - Cookie value doesn't match message hash.
+        if (!this.state.shownMessage && (!hash || hash !== data.hash)) {
+          this.state.shownMessage = msg;
```

Comment:
Are you trying to pick a message object from the children passed to this component? Would it make more sense to require the child be singular, and of the correct type in the prop type checking instead?